app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt

```diff
@@ -427,6 +429,105 @@ class ProfileEditActivityTest {
     }
   }
 
+  @Test
+  fun testProfileEdit_startWithUserHasDownloadAccess_clickAllowDownloadContainer_checkChanged() {
+    profileManagementController.addProfile(
+      name = "James",
+      pin = "123",
+      avatarImagePath = null,
+      allowDownloadAccess = true,
+      colorRgb = -10710042,
+      isAdmin = false
+    ).toLiveData()
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 4
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_edit_allow_download_switch)).check(matches(isChecked()))
+      onView(withId(R.id.profile_edit_allow_download_container)).perform(click())
+      onView(withId(R.id.profile_edit_allow_download_switch)).check(matches(not(isChecked())))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserHasDownloadAccess_switchIsNotClickable() {
+    profileManagementController.addProfile(
+      name = "James",
+      pin = "123",
+      avatarImagePath = null,
+      allowDownloadAccess = true,
```

Comment:
having download access but not able to click? is this correct?

Reply:
Yes, we are just checking that `Switch` is not clickable instead the entire container is clickable which is written as test in above `testProfileEdit_startWithUserHasDownloadAccess_clickAllowDownloadContainer_checkChanged `

The aim is to make `Switch` non-clickable and `Switch + 2 TextViews` as clickable.

Reply:
@anandwana001 PTAL

Reply:
my question was more of logic one, when we provide `allowDownloadAccess ` to true then is it correct that it is not clickable?

Reply:
`allowDownloadAccess` has nothing to do with `clickable`. If its true, it means that while creating the profile user has given the download access to user and therefore show the switch as `ON` by default which the admin can further change on click in this screen.

`allowDownloadAccess` is just the state of the permission to user and not related to click.

The click will simply change this state.